trunk/SUAVE/Methods/Aerodynamics/Common/Fidelity_Zero/Lift/generate_vortex_distribution.py

```diff
@@ -0,0 +1,1413 @@
+            
+            for cs_ID in cs_IDs[cs_IDs >= 0]:
+                cs_tag     = wing.tag + '__cs_id_{}'.format(cs_ID)
+                cs_wing    = wings[cs_tag]
+                rel_offset = cs_wing.origin[0,1] if not vertical_wing else cs_wing.origin[0,2]
+                cs_wing.y_coords_required.append(y_coord - rel_offset)
+            
+            if y_coordinates[idx_y+1] == break_spans[i_break+1]: 
+                i_break += 1
+            
+    
+    # -------------------------------------------------------------------------------------------------------------
+    # Run the stip contruction loop again if wing is symmetric. Reflection plane = x-y plane for vertical wings. Otherwise, reflection plane = x-z plane
+    signs = np.array([1, -1]) # acts as a multiplier for symmetry. -1 is only ever used for symmetric wings
+    for sym_sign in signs[[True,sym_para]]:
+        # create empty vectors for coordinates 
+        xah   = np.zeros(n_cw*n_sw)
+        yah   = np.zeros(n_cw*n_sw)
+        zah   = np.zeros(n_cw*n_sw)
+        xbh   = np.zeros(n_cw*n_sw)
+        ybh   = np.zeros(n_cw*n_sw)
+        zbh   = np.zeros(n_cw*n_sw)    
+        xch   = np.zeros(n_cw*n_sw)
+        ych   = np.zeros(n_cw*n_sw)
+        zch   = np.zeros(n_cw*n_sw)    
+        xa1   = np.zeros(n_cw*n_sw)
+        ya1   = np.zeros(n_cw*n_sw)
+        za1   = np.zeros(n_cw*n_sw)
+        xa2   = np.zeros(n_cw*n_sw)
+        ya2   = np.zeros(n_cw*n_sw)
+        za2   = np.zeros(n_cw*n_sw)    
+        xb1   = np.zeros(n_cw*n_sw)
+        yb1   = np.zeros(n_cw*n_sw)
+        zb1   = np.zeros(n_cw*n_sw)
+        xb2   = np.zeros(n_cw*n_sw) 
+        yb2   = np.zeros(n_cw*n_sw) 
+        zb2   = np.zeros(n_cw*n_sw)    
+        xac   = np.zeros(n_cw*n_sw)
+        yac   = np.zeros(n_cw*n_sw)
+        zac   = np.zeros(n_cw*n_sw)    
+        xbc   = np.zeros(n_cw*n_sw)
+        ybc   = np.zeros(n_cw*n_sw)
+        zbc   = np.zeros(n_cw*n_sw)     
+        xa_te = np.zeros(n_cw*n_sw)
+        ya_te = np.zeros(n_cw*n_sw)
+        za_te = np.zeros(n_cw*n_sw)    
+        xb_te = np.zeros(n_cw*n_sw)
+        yb_te = np.zeros(n_cw*n_sw)
+        zb_te = np.zeros(n_cw*n_sw)  
+        xc    = np.zeros(n_cw*n_sw) 
+        yc    = np.zeros(n_cw*n_sw) 
+        zc    = np.zeros(n_cw*n_sw) 
+        x     = np.zeros((n_cw+1)*(n_sw+1)) # may have to change to make space for split if control surfaces are allowed to have more than two Segments
+        y     = np.zeros((n_cw+1)*(n_sw+1)) 
+        z     = np.zeros((n_cw+1)*(n_sw+1))         
+        cs_w  = np.zeros(n_sw)        
+        
+        # ---------------------------------------------------------------------------------------------------------
+        # Loop over each strip of panels in the wing
+        i_break = 0           
+        for idx_y in range(n_sw):
+            # define basic geometric values------------------------------------------------------------------------
```

Comment:
Can this loop be vectorized?

Reply:
probably, but I think that would be a good thing to put in another PR. Since control surfaces have different discretizations from the wings, this could throw a wrench in the vectorization that probably deserve a separate review. Plus, the discretization function runs in a fraction of the time that the full VLM runs.